pkg/sql/catalog/fetchpb/index_fetch.proto

```diff
@@ -130,13 +131,13 @@ message IndexFetchSpec {
 
   message ExternalRowData {
     optional util.hlc.Timestamp as_of = 1 [(gogoproto.nullable) = false];
-    // OldPrefix contains the original TenantPrefix + TableID prefix of the keys
-    // produced by SQL layer that need to be remapped to the external data.
-    optional bytes old_prefix = 2;
-    // NewPrefix contains the re-mapped TenantPrefix + TableID prefix of the
-    // keys that need to be used against the KV layer to fetch the external
-    // data.
-    optional bytes new_prefix = 3;
+    // TenantID is the new tenant ID to use in keys when fetching external data
```

Comment:
actually, you know, this will just error out with wrong wire type if we do this in place which is maybe nice (and loud) compared to the alternative which would be a gate. Carry on.